packages/account/src/Sections/Security/TwoFactorAuthentication/digit-form.tsx

```diff
@@ -1,64 +1,74 @@
 import React from 'react';
 import classNames from 'classnames';
-import { Formik, Form, Field } from 'formik';
+import { Formik, Form, Field, FieldProps, FormikProps } from 'formik';
 import { Input, Button } from '@deriv/components';
-import { getPropertyValue, WS } from '@deriv/shared';
+import { useSendUserOTP } from '@deriv/api';
 import { localize } from '@deriv/translations';
+import { useStore } from '@deriv/stores';
 
-const DigitForm = ({ is_enabled, setTwoFAStatus, setTwoFAChangedStatus, is_language_changing }) => {
-    const [is_success, setSuccess] = React.useState(false);
-    const [is_ready_for_verification, setReadyForVerification] = React.useState(false);
-    const button_text = is_enabled ? localize('Disable') : localize('Enable');
-    const formik_ref = React.useRef();
-    let enable_response;
+type TDigitFormValues = {
+    digit_code: string;
+};
+
+const DigitForm = () => {
+    const { client, common } = useStore();
+    const { is_language_changing } = common;
+    const { has_enabled_two_fa, setTwoFAChangedStatus, setTwoFAStatus } = client;
+
+    const { is_TwoFA_enabled, error, isSuccess, sendUserOTP } = useSendUserOTP();
+    const button_text = has_enabled_two_fa ? localize('Disable') : localize('Enable');
+    const formik_ref = React.useRef<FormikProps<TDigitFormValues>>(null);
 
     const initial_form = {
         digit_code: '',
     };
 
     React.useEffect(() => {
         if (is_language_changing) {
-            formik_ref.current.setFieldTouched('digit_code');
+            formik_ref.current?.setFieldTouched('digit_code');
         }
     }, [is_language_changing]);
 
-    const validateFields = async values => {
+    React.useEffect(() => {
+        if (error) {
+            if (typeof error === 'object' && 'code' in error && 'message' in error) {
+                const { code, message } = error;
+                if (code === 'InvalidOTP')
+                    formik_ref.current?.setFieldError(
+                        'digit_code',
+                        localize("That's not the right code. Please try again.")
+                    );
+                else {
+                    formik_ref.current?.setFieldError('digit_code', message as string);
+                }
+            }
+        }
+    }, [error]);
+
+    React.useEffect(() => {
+        if (isSuccess) {
+            setTwoFAStatus(is_TwoFA_enabled);
+            setTwoFAChangedStatus(true);
+            formik_ref.current?.resetForm();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do we disable this?

Reply:
Because ts requests to add setTwoFAStatus and setTwoFAChangedStatus also in dependencies but I feel they are not required.